### PR TITLE
Increase coverage for IT tests

### PR DESCRIPTION
**Pull request by NikitaZhevnitskiy**
1. Separate reports for IT and unit tests
2. Added more tests

ps. I would like to receive feedback @jeqo, are there any priority for feature to test first ?

### CODE REVIEW

File: storage/src/test/java/zipkin2/storage/kafka/KafkaStorageIT.java [hunk 1]
```diff
@@ -167,4 +167,302 @@ public void shouldFindTraces() throws Exception {
...
+
+    List<Span> spans = Arrays.asList(span1, span2);
+    storage.spanConsumer().accept(spans).execute();
+
+    IntegrationTestUtils.waitUntilMinRecordsReceived(
+        testConsumerConfig, storage.spansTopic.name, 2, 10000);
+
+    // query by annotation {"key_tag_a":"value_tag_a"} = 1 trace
+    await()
+        .pollDelay(5, TimeUnit.SECONDS)
```
Comment (jeqo) [inline storage/src/test/java/zipkin2/storage/kafka/KafkaStorageIT.java]: Isn't the `.atMost(5, TimeUnit.SECONDS)` a better match for this case?

We are waiting for the store to pick up the change on the output topic, after we have validated that messages are published on the topic L210.
If we are validating that message is already published, then any moment, from that point, the storage could be updated.

Using `atMost` we are telling to check condition during 5 secs, and if not match, then fail.
With `pollDelay` we are waiting 5secs and then validating the condition.

It seems that the first one covers better the eventual consistency on this operation. 

WDYT?

File: storage/src/test/java/zipkin2/storage/kafka/KafkaStorageIT.java [hunk 2]
```diff
@@ -167,4 +167,302 @@ public void shouldFindTraces() throws Exception {
...
+                              .limit(10)
+                              .lookback(Duration.ofMinutes(1).toMillis())
+                              .build())
+                      .execute();
+              return traces.size() == 1;
+            });
+
+    // query by annotation {"key_tag_non_exist_a":"value_tag_non_exist_a"} = 1 trace
+    await()
+        .pollDelay(5, TimeUnit.SECONDS)
```
Comment (jeqo) [inline storage/src/test/java/zipkin2/storage/kafka/KafkaStorageIT.java]: Same here.

File: storage/src/test/java/zipkin2/storage/kafka/KafkaStorageIT.java [hunk 3]
```diff
@@ -167,4 +167,302 @@ public void shouldFindTraces() throws Exception {
...
+
+    List<Span> spans = Arrays.asList(span1, span2);
+    storage.spanConsumer().accept(spans).execute();
+
+    IntegrationTestUtils.waitUntilMinRecordsReceived(
+        testConsumerConfig, storage.spansTopic.name, 2, 10000);
+
+    // query by span name `op_a` = 1 trace
+    await()
+        .atMost(5, TimeUnit.SECONDS)
```
Comment (jeqo) [inline storage/src/test/java/zipkin2/storage/kafka/KafkaStorageIT.java]: same here.

File: storage/src/test/java/zipkin2/storage/kafka/KafkaStorageIT.java [hunk 4]
```diff
@@ -167,4 +167,302 @@ public void shouldFindTraces() throws Exception {
...
+                              .limit(10)
+                              .lookback(Duration.ofMinutes(1).toMillis())
+                              .build())
+                      .execute();
+              return traces.size() == 1;
+            });
+
+    // query by span name `op_b` = 1 trace
+    await()
+        .atMost(5, TimeUnit.SECONDS)
```
Comment (jeqo) [inline storage/src/test/java/zipkin2/storage/kafka/KafkaStorageIT.java]: Same here.

File: storage/src/test/java/zipkin2/storage/kafka/KafkaStorageTest.java [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+      builder.indexStorageDirectory(null);
+      fail();
+    } catch (NullPointerException ignored) {
+    }
+  }
+
+  @Test
+  public void topicDefault() {
+      try{
+          KafkaStorage.Topic.Builder topicBuilder = KafkaStorage.Topic.builder(null);
```
Comment (jeqo) [inline storage/src/test/java/zipkin2/storage/kafka/KafkaStorageTest.java]: nit: remove unused `topicBuilder` variable.

File: storage/src/test/java/zipkin2/storage/kafka/KafkaStorageIT.java (hunk 1 above)
Comment (NikitaZhevnitskiy) [reply storage/src/test/java/zipkin2/storage/kafka/KafkaStorageIT.java]: agree.

File: storage/src/test/java/zipkin2/storage/kafka/KafkaStorageIT.java (hunk 2 above)
Comment (NikitaZhevnitskiy) [reply storage/src/test/java/zipkin2/storage/kafka/KafkaStorageIT.java]: There are some cases, where it should not found traces (f.ex. with non_existing_service_name), in that case exactly need delayPoll.
I will update comment

**Comment by jeqo** [comment]
LGTM thanks @NikitaZhevnitskiy !

**Pull request merged**